From 4b5dc596547e0833e2cccf65efa46ecbe7d0b99c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?G=C3=B6kt=C3=BCrk=20Y=C3=BCksek?= <gokturk@gentoo.org>
Date: Fri, 6 Jul 2018 14:41:52 -0400
Subject: [PATCH] Makefile: add support for $(DESTDIR) in install target

Allow install target to install into an arbitrary root instead of '/'
by prefixing $(PREFIX) with $(DESTDIR).
---
 Makefile | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index c38b800..2c778f2 100644
--- a/Makefile
+++ b/Makefile
@@ -44,11 +44,13 @@ scan:	$(OBJS)
 	scan-build --use-analyzer=/usr/bin/clang $(CC) -shared -Wl,-soname,lib$(NAME).so.$(LIBMAJOR) -o lib$(NAME).so.$(LIBVERSION) $(OBJS) $(LDFLAGS)
 
 install:
-	install -m 644 doc/$(NAME).3 $(PREFIX)/share/man/man3/
-	gzip -9 $(PREFIX)/share/man/man3/$(NAME).3
-	install -m 0755 -s lib$(NAME).so.$(LIBVERSION) $(PREFIX)/$(LIBDIR)/
-	$(RM) $(PREFIX)/$(LIBDIR)/lib$(NAME).so.$(LIBMAJOR)
-	ln -s lib$(NAME).so.$(LIBVERSION) $(PREFIX)/$(LIBDIR)/lib$(NAME).so.$(LIBMAJOR)
+	install -d -m 0755 $(DESTDIR)$(PREFIX)/share/man/man3
+	install -m 644 doc/$(NAME).3 $(DESTDIR)$(PREFIX)/share/man/man3/
+	gzip -9 $(DESTDIR)$(PREFIX)/share/man/man3/$(NAME).3
+	install -d -m 0755 $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	install -m 0755 -s lib$(NAME).so.$(LIBVERSION) $(DESTDIR)$(PREFIX)/$(LIBDIR)/
+	$(RM) $(DESTDIR)$(PREFIX)/$(LIBDIR)/lib$(NAME).so.$(LIBMAJOR)
+	ln -s lib$(NAME).so.$(LIBVERSION) $(DESTDIR)$(PREFIX)/$(LIBDIR)/lib$(NAME).so.$(LIBMAJOR)
 
 clean:
 	@- $(RM) $(NAME)
-- 
2.17.0

